examples/pattern-matcher/README.md

```diff
@@ -0,0 +1,19 @@
+Pattern Matcher Examples
+------------------------
+
+See `../guile/README` for for an intro on running the guile shell needed
+to run the demos here.
+
+The first set of demos provide a basic introduction to various
+main features.
+
+* simple.scm: a basic introduction to pattern matching.
+* gsn.scm: calling arbitrary functions upon a match.
+* gpn.scm: calling arbitrary functions to decide a match.
+
+Some simpler applications showing some things one can do, and how to do
+them.
+
+* fsm-basic.scm: A simple Finite State Machine demo.
+* fsm-full.scm: A generic finite-state-machine constructor
+* markov-chain.scm: A Markov chain (probabilistic FSM) based on above.
```

Comment:
I forgot to push DeleteLink.cc and the demos won't work without it. I just pushed; they should now work.